File: frontend/src/api/course.js
import http from '@/utils/http'

export function loginAPI(params) {
    /* @param stuId
    *  @param password
    *  @param termId  */
    return http.post('/course/login', params)
}

export function readCourseRankAPI(params) {
    return http.get('/course/course_rank', params)
}
export function readCourseInfoAPI(params) {
    return http.get('/course/course_info', params)
}
export function readCourseTableAPI(params) {
    return http.get('/course/course_table', params)
}
export function readStuInfoAPI(params) {
    return http.get('/course/stu_info', params)
}

File: backend/app/routers/api_v1/course.py
from typing import List
from fastapi import APIRouter
from fastapi import Depends, FastAPI, HTTPException
from sqlalchemy.orm import Session
from fastapi.middleware.cors import CORSMiddleware
# from utils import logger
from schemas.course import *
from utils import CourseHelper


router = APIRouter()


@router.post('/login', tags=['login'])
def login(login_info: LoginInfo):
    global xk
    xk = CourseHelper(login_info.stu_id, login_info.password, login_info.term_id)
    return {'login_state': xk.login_state}


@router.get('/course_rank',  response_model=List[CourseRank], tags=['Get SHU course rank'])
def get_course_rank():
    return xk.switch2dict(xk.get_course_rank())


@router.get('/course_info', response_model=List[CourseInfo], tags=['Get SHU course information'])
def get_course_info():
    return xk.switch2dict(xk.get_course_info())


@router.get('/course_table',  response_model=List[CourseTable], tags=['Get SHU course table'])
def get_course_table():
    return xk.switch2dict(xk.get_course_table())


@router.get('/stu_info',  response_model=StuInfo, tags=['Get SHU self info'])
def get_stu_info():
    return xk.switch2dict(xk.get_stu_info())[0]
